### ISSUE TITLE
Quicksilver retains focus after triggering different app

### ISSUE THREAD

**Issue by jaygooby** [quicksilver/Quicksilver#2596]
For questions or tech support, please start with the [users’ group][1]. If you know the problem is in Quicksilver itself, report it here.

[1]: https://groups.google.com/forum/#!forum/blacktree-quicksilver

**Describe the bug**
I'm on an M1 Mac, macOS Monterey 12.1

Quicksilver is configured to not appear in Dock, but in the menu bar:

<img width="860" alt="Screenshot 2022-02-03 at 11 47 19" src="https://user-images.githubusercontent.com/64270/152337182-19b1c73c-ec44-4f95-9469-e835b73c1cc8.png">

If I trigger even something simple like launching Safari, Quicksilver seems to retain focus, so e.g. moving between tabs in Safari using shit-cmd-arrow doesn't work. You need to alt-tab Safari into the foreground before the app really has focus.

**To Reproduce**
1. Have Safari open with a number of tabs
2. Use Quicksilver to open Safari 
3. Use keyboard to navigate between Safari tabs - nothing happens (aside from the "boink" noise)
4. Alt-tab away from and then back to Safari - tab navigation now works

**Expected behavior**
When Quicksilver launches or opens an application, that application has keyboard focus

**What version of macOS?**
Monterey 12.1 on M1 Mac

**What version of Quicksilver?**
1.6.1

**Comment by n8henrie** [comment]
Trying to reproduce, also on an M1 MBP, 12.2, 1.6.1 -- are you sure this is a Quicksilver problem and not just Safari? Full-screen or not (or both)?

For example, if I open two tabs in Safari, neither of which has a search box:

- https://qsapp.com
- https://n8henrie.com

I can switch back to Firefox, type into this form, use QS to go back to Safari, and e.g. <kbd>kbd</kbd><kbd>shift</kbd><kbd><-</kbd> switches tabs as expected.

However, if I open tabs that include a focus-stealing search box:

- https://www.google.com
- https://www.yahoo.com

Either of these steals my focus (sometimes?) and <kbd>kbd</kbd><kbd>shift</kbd><kbd><-</kbd> no longer works.

**Comment by jaygooby** [comment]
Ah, thanks for the those tests. It's helped me narrow the problem down:

1. Give Safari focus (with your two non-focus stealing tab; https://qsapp.com/ and https://n8henrie.com/) 
2. Trigger Quicksilver
3. Press escape to cancel the bezel

Safari looks like it has focus, but it does't - you still have to alt-tab away to another app and back again, before it does.

**Comment by n8henrie** [comment]
> Safari looks like it has focus, but it does't

Interesting. In that case, I can't reproduce.

- I have QS in the menu bar enabled
- I'm typing this into Firefox
- QS -> `Safari` -> `Open`
- I can cmd-shift-arrow between tabs as expected, immediately, without having to click anything to give focus to Safari.
- This is true in both full-screen and non-fullscreen

**Comment by jaygooby** [comment]
Try this:

1. Have Firefox focused
2. Trigger QS
3. Press escape
4. Try and use your keyboard to navigate tabs in Firefox

The problem seems to be dropping back to the app that had focus before QS is cancelled

**Comment by pjrobertson** [comment]
I also tried reproducing this, but cannot find the issue. @jaygooby - can you please try and debug further. Also, sending screenshots of each of your preference panes may help

https://user-images.githubusercontent.com/150431/152452886-2d0cf088-6b38-4d59-8b28-7512b88cb417.mov

**Comment by jaygooby** [comment]
Weird. Thanks. I wondered if it might be due to another program (had iTerm and LastPass both in the menu bar), but I've quit everything and still no joy. This screencast uses the mic, so you can hear the boink when I try and switch between the tabs

https://user-images.githubusercontent.com/64270/152575205-337e54a1-cb62-4def-ac0b-744b1a258229.mov

**Comment by jaygooby** [comment]
Here are the grabs from the pref panes:
<img width="860" alt="Screenshot 2022-02-04 at 17 32 01" src="https://user-images.githubusercontent.com/64270/152575713-00e383dc-199b-49bc-9da4-24cb6826d6d4.png">

<img width="860" alt="Screenshot 2022-02-04 at 17 32 07" src="https://user-images.githubusercontent.com/64270/152575737-214a4bdf-ad7e-4a7c-8e1c-1a58d7295a49.png">

<img width="860" alt="Screenshot 2022-02-04 at 17 32 12" src="https://user-images.githubusercontent.com/64270/152575758-45fcf1b7-335e-4cc2-8fad-1da07a062577.png">

<img width="860" alt="Screenshot 2022-02-04 at 17 32 18" src="https://user-images.githubusercontent.com/64270/152575794-e068cc5c-f8da-4137-886c-a6a7300ea309.png">

<img width="860" alt="Screenshot 2022-02-04 at 17 32 26" src="https://user-images.githubusercontent.com/64270/152575841-f6d25f29-f4ed-4c92-9112-8050eaad5138.png">

<img width="860" alt="Screenshot 2022-02-04 at 17 32 34" src="https://user-images.githubusercontent.com/64270/152575873-cc6dc447-b1f3-45f2-b2ea-372a0ae1a5fb.png">

**Comment by n8henrie** [comment]
> Try this:
>
>    Have Firefox focused
    Trigger QS
    Press escape
    Try and use your keyboard to navigate tabs in Firefox

@jaygooby this works flawlessly for me, with and without fullscreen, in the menu bar. Even tried restarting quicksilver to see if that made a difference.

**Comment by n8henrie** [comment]
Are you running an ARM64 build of QS? Or still on x86_64? I wonder if that could make a difference (I'm on the former).

**Comment by pjrobertson** [comment]
Can you try deselecting "Use superfluous visual effects" and "focus action
when showing results" one by one to see if that has any effect?

**Comment by skurfer** [comment]
Sounds similar to #1351, but I can’t reproduce it as described either. Only setting that’s likely to matter here is “Reopening the current application”, but I have the same value as shown in the screenshot.

**Comment by jaygooby** [comment]
@n8henrie it's a `brew --cask` installed version. Interesting, I checked in Activity Monitor and it's showing as intel. This is the current cask recipe: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/quicksilver.rb

**Comment by jaygooby** [comment]
@pjrobertson - tried both of those, one at a time, and restarted after each, but no joy

**Comment by jaygooby** [comment]
I'm going to close this now as I can no longer reproduce either. There have been a couple of Quicksilver and plugin updates since I opened the issue. I'm now on QS 2.1.0 (4032).

Sorry to not be more help :(

**Comment by n8henrie** [comment]
@jaygooby thinking back, I may have run into this a time or two in the interim -- if I remember correctly, it was when trying to use the `Type Text` action (and/or `Paste as Plain Text`), which sadly hadn't worked for me for months, and I use heavily (mostly to get around password forms that have pasting blocked, since I use a password manager). Whenever I tried to use this action it would fail, and around that time I would often end up in a similar situation to what you describe -- no app would seem to have focus.

Anyway, last week I finally figured out that there was an issue in System Preferences -> Security and Privacy -> Accessibility (something along those lines); removing the entry for Quicksilver and then invoking `Type Text` caused the accessibility permissions prompt to reappear, and after re-enabling it for QS these actions now work perfectly, and I can't reproduce the losing focus bug.

If it comes back, may be something to think about.
